docs/dev/replay-exec.md

```diff
@@ -0,0 +1,82 @@
+This doc describes how to dump an Exec runtime meta and column batch data to a local directory,
+and replay the Exec with the dumped meta and data. When encountering a perf issue using this tool
+can dump the runtime meta/data and then replay. It will spend less time when collecting NSYS and 
+NCU information when replaying the dumped runtime meta/data.
+
+# Dump a Exec meta and a column batch data
+
+## compile Spark-Rapids jar
+e.g.: compile for Spark330
+```
+mvn clean install -DskipTests -pl dist -am -DallowConventionalDistJar=true -Dbuildver=330 
+```
+
+## enable dump
+This assumes that the RAPIDs Accelerator has already been enabled.   
+This feature is disabled by default.   
+Set the following configurations to enable this feature:   
+
+``` 
+spark.conf.set("spark.rapids.sql.test.replay.exec.type", "project")
+```
+Default `type` value is empty which means do not dump.   
+Set this `type` to `project` if you want to dump Project Exec runtime data. Currently only support
+`project` and empty.
+
+```
+spark.conf.set("spark.rapids.sql.test.replay.exec.dumpDir", "file:/tmp")
+```
+Default value is `file:/tmp`.  
+specify the dump directory, e.g.: `file:/tmp/my-debug-path`.  
+This path should be a directory or someplace that we can create a directory to
+store files in. Remote path is supported e.g.: `hdfs://url:9000/path/to/save`
+
+```
+spark.conf.set("spark.rapids.sql.test.replay.exec.threshold.timeMS", 100)
+```
+Default value is 100MS.   
+Only dump the column batches when it's executing time exceeds threshold time. 
+
+```  
+spark.conf.set("spark.rapids.sql.test.replay.exec.maxBatchNum", 1)
```

Comment:
nit: will be more suitable saying `spark.rapids.sql.test.replay.batch.limit` and mention it's per executor base or task.

Reply:
Done.